addons/crm/models/res_partner.py

```diff
@@ -77,19 +77,19 @@ def _compute_meeting(self):
             meeting_data = self.env.cr.fetchall()
 
             # Create a dict {partner_id: event_ids} and fill with events linked to the partner
-            meetings = {p.id: set() for p in all_partners}
+            meetings = {k: set() for k in set(m[0] for m in meeting_data)}
             for m in meeting_data:
                 meetings[m[0]].add(m[1])
 
             # Add the events linked to the children of the partner
-            all_partners.read(['parent_id'])
-            for p in all_partners:
+            partners_with_meetings = self.browse(meetings.keys())
+            for p in partners_with_meetings:
                 partner = p
-                while partner:
+                while partner.parent_id:
+                    partner = partner.parent_id
                     if partner in self:
                         meetings[partner.id] |= meetings[p.id]
-                    partner = partner.parent_id
```

Comment:
Was a bit confused when I first looked at it but I guess we don't need to do this for the partner itself since it will just do nothing.. Variable names are a bit confusing ^^'

Also I guess the `while` might be superfluous here? I don't think you can have multiple levels of children for contacts. And if you can the method seems like it wouldn't work very well (we'd need to update the meetings level by level)

Reply:
Yes I tried to leave the variable names like they were as much as possible to avoid a bigger diff but I can still rename them if you think it's better.
I think multiple layers of children in the contact are not that uncommon since the OPW from which originates this PR had some. Therefore I think the `while` loop is necessary so that each meeting is reflected to the top parent. But let me know if you think there is an more efficient way to do it :wink:

Reply:
I see. It would indeed make sense in some rare cases. But like you said I think it adds a bit much overhead for the majority of cases. If it's ok for you I think we can keep it as it is for the time being :wink: